providers/implementations/kem/rsa_kem.c

```diff
@@ -329,13 +347,24 @@ static int rsasve_recover(PROV_RSA_CTX *prsactx,
         return 1;
     }
 
-    /* Step (2): check the input ciphertext 'inlen' matches the nlen */
+    /*
+     * Step (2): check the input ciphertext 'inlen' matches the nlen
+     * and that outlen is at least nlen bytes
+     */
     if (inlen != nlen) {
         ERR_raise(ERR_LIB_PROV, PROV_R_BAD_LENGTH);
         return 0;
     }
+    if (outlen == NULL || *outlen < nlen) {
```

Comment:
since you are adding `outlen == NULL` test here, I wonder if the similar check should be added to existing code above (line 346) in new code.

Reply:
I believe @t-j-h mentioned the same to me.  Will fix

Reply:
resolved